### PR TITLE
improve corner case handling of sed::integrate

### PR DESCRIPTION
**Pull request by johannct**
improve slightly the corner cases and add test

## Change Description

- [x] My PR includes a link to the issue that I am addressing

## Code Quality
- [ ] I have read the Contribution Guide
- [ ] My code follows the code style of this project
- [ ] My code builds (or compiles) cleanly without any errors or warnings
- [ ] My code contains relevant comments and necessary documentation

### Bug Fix Checklist
- [ ] My fix includes a new test that breaks as a result of the bug (if possible)
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### New Feature Checklist
- [ ] I have added or updated the docstrings associated with my feature using the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover my new feature
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### Documentation Change Checklist
- [ ] Any updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)

### Build/CI Change Checklist
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the README to reflect this
- [ ] If this is a new CI setup, I have added the associated badge to the README

### Other Change Checklist
- [ ] Any new or updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover any changes
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### CODE REVIEW

File: src/lib/SED.cpp [hunk 1]
```diff
@@ -268,6 +272,10 @@ double SED::integrate(const double lmin, const double lmax) {
     double dlbd = (lamb_flux[i + 1].lamb - lamb_flux[i].lamb);
     res += dlbd * fmean;
   }
+
+  // Clamp lastidx to valid range - single comparison, negligible cost
+  lastidx = std::min(lastidx, (size_t)(lamb_flux.size() - 1));
+
   // integrate from lamb_flux[lastidx] to lmax
   x1 = lamb_flux[lastidx].lamb;
   x2 = lamb_flux[lastidx + 1].lamb;
```
[the remark this answers is not in the document]
Comment (johannct) [reply src/lib/SED.cpp]: this piece of code is removed, as not useful anyway

**Review by olivierilbert — APPROVED** [review]
good for me
